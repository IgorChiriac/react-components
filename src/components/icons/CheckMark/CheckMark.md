Wrapper around svg code for a checkmark icon.

```jsx
import { CheckMarkIcon } from '@zopauk/react-components';

<CheckMarkIcon />;
```
