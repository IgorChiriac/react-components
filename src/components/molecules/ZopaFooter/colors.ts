export const neutral900 = '#1C2139';
export const neutral200 = '#A9ACBA';
export const neutral400 = '#6F748B';
export const neutral800 = '#282F52';
