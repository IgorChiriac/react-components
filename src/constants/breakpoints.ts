export const breakpoints = {
  desktop: 1280,
  phone: 600,
  tablet: 720,
} as const;

export type TDeviceSizes = keyof typeof breakpoints;
