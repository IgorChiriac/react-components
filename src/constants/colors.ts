export type TTextHexColors = '#FFFFFF' | '#C5C6CD' | '#0D0A38' | '#1EC06A' | '#EE0505';
export type THeadingHexColors = '#FFFFFF' | '#0D0A38';
export type TLinkHexColors = '#FFFFFF' | '#0D0A38';

export const colors = {
  base: {
    primary: '#00B9A7',
    secondary: '#4A3EDE',
  },
  neutral: {
    white: '#FFFFFF',
    light: '#EAEAEE',
    medium: '#C5C6CD',
    dark: '#0D0A38',
  },
  semantic: {
    success: '#1EC06A',
    alert: '#FFB428',
    error: '#EE0505',
  },
};
