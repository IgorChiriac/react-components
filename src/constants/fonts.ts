export const openSans: string =
  '"Open Sans", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"';
export const alverata: string = 'Alverata, "Arial Black", Verdana, sans-serif';
