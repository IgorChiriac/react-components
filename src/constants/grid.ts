const grid = {
  breakpoints: {
    l: 992,
    m: 768,
    s: 576,
    xl: 1200,
    xs: 0,
  },
  cols: 12,
  gutter: 16,
  width: {
    l: 960,
    m: 720,
    s: 540,
    xl: 1140,
  },
};

export default grid;
