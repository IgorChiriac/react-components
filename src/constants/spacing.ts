export const spacing = [0, 4, 8, 16, 24, 32, 40, 56, 64, 104];
