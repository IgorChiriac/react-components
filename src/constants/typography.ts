const systemFontStack =
  '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"';

const typography = {
  primary: `"Open Sans",${systemFontStack}`,
  sizes: {
    text: {
      base: '16px',
      small: '14px',
    },
    heading: {
      h1: '48px',
      h2: '28px',
      h3: '24px',
      h4: '20px',
      h5: '20px',
      h6: '20px',
    },
  },
  weights: {
    regular: 400,
    semibold: 600,
    bold: 700,
  },
  lineHeights: {
    text: 1.6,
    heading: 1.1,
  },
};

export { typography };
