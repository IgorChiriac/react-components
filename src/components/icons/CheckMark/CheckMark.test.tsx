import React from 'react';
import { render } from '@testing-library/react';
import * as colors from '../../../constants/colors';
import CheckMark from './CheckMark';

describe('<CheckMark />', () => {
  it(`renders without crashing`, () => {
    const { container } = render(<CheckMark />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it(`renders with a specific color`, () => {
    const { container } = render(<CheckMark color={colors.primary.pink500} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
