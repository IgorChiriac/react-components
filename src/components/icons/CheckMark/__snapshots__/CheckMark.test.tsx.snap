// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckMark /> renders with a specific color 1`] = `
<svg
  aria-label="Checkmark"
  height="12px"
  role="img"
  version="1.1"
  viewBox="0 0 14 12"
  width="14px"
>
  <g
    fill="none"
    fill-rule="evenodd"
    stroke="none"
    stroke-width="1"
    transform="translate(-5.000000, -7.000000)"
  >
    <g
      fill="#1EC06A"
    >
      <path
        d="M8.8937112,18.1162104 L5.28067153,13.7796715 C4.86185798,13.276869 4.91785534,12.5208652 5.40783219,12.088863 C5.89547582,11.6556608 6.63277432,11.7132611 7.05275448,12.2196637 L9.70329594,15.3996797 L16.9806188,7.38003928 C17.4204314,6.89643685 18.1588965,6.87123672 18.629041,7.32243899 C19.100352,7.77484127 19.1260175,8.53324509 18.6862049,9.01924754 L10.4025549,18.1480964 C10.0314235,18.5570951 9.3990031,18.5877927 8.99000439,18.2166613 C8.95561314,18.1854542 8.92343754,18.1518893 8.8937112,18.1162104 Z"
      />
    </g>
  </g>
</svg>
`;

exports[`<CheckMark /> renders without crashing 1`] = `
<svg
  aria-label="Checkmark"
  height="12px"
  role="img"
  version="1.1"
  viewBox="0 0 14 12"
  width="14px"
>
  <g
    fill="none"
    fill-rule="evenodd"
    stroke="none"
    stroke-width="1"
    transform="translate(-5.000000, -7.000000)"
  >
    <g
      fill="#4A3EDE"
    >
      <path
        d="M8.8937112,18.1162104 L5.28067153,13.7796715 C4.86185798,13.276869 4.91785534,12.5208652 5.40783219,12.088863 C5.89547582,11.6556608 6.63277432,11.7132611 7.05275448,12.2196637 L9.70329594,15.3996797 L16.9806188,7.38003928 C17.4204314,6.89643685 18.1588965,6.87123672 18.629041,7.32243899 C19.100352,7.77484127 19.1260175,8.53324509 18.6862049,9.01924754 L10.4025549,18.1480964 C10.0314235,18.5570951 9.3990031,18.5877927 8.99000439,18.2166613 C8.95561314,18.1854542 8.92343754,18.1518893 8.8937112,18.1162104 Z"
      />
    </g>
  </g>
</svg>
`;
